.changeset/funny-paws-dream.md

```diff
@@ -0,0 +1,30 @@
+---
+"@comet/cms-admin": minor
+---
+
+Add `ContentScopeSelect` component
+
+This can be used as the basis for both content-driven and data-driven applications.
+
+**Example**
+
+```tsx
+function ContentScopeControls() {
+    const [value, setValue] = useState({ domain: "main", language: "en" });
+
+    return (
+        <ContentScopeSelect
+            value={value}
+            onChange={(value) => {
+                setValue(value);
+            }}
```

Comment:
I think this can be simplified:

```suggestion
            onChange={setValue}
```

Reply:
I first had it like this, but then the type inference for `groupBy` didn't work correctly:

<img width="1414" alt="image" src="https://github.com/vivid-planet/comet/assets/48853629/c8378f47-d13d-4556-b5c8-a472cb69b76a">

Also, while passing the `Dispatch<SetStateAction<S>>` works, it's not very "clean" IMO, since we assume that setState returns `void`, which may change in a future React version.